test/functional/api/v1/searches_controller_test.rb

```diff
@@ -53,16 +53,25 @@ def self.should_respond_to(format)
     end
 
     context "with elasticsearch down" do
-      should "fallback to legacy search" do
+      should "returns friendly error message" do
         requires_toxiproxy
         Toxiproxy[:elasticsearch].down do
           get :show, params: { query: "other" }, format: :json
 
-          assert_response :success
-          assert_equal "other", JSON.parse(@response.body).first["name"]
+          assert_response :service_unavailable
+          assert_equal "search not available, please try again later", @response.body
         end
       end
     end
+
+    context "invalid query" do
+      should "returns friendly error message" do
+        get :show, params: { query: "AND other" }, format: :json
+
+        assert_response :bad_request
+        assert_equal "invalid query", @response.body
```

Comment:
can we bubble up the error message from the opensearch cluster here? it might be more helpful to the user than just saying "invalid"

Reply:
this is example of error, I'm not sure there is anything interesting to share in response :thinking: 

```
#<Searchkick::InvalidQueryError: [400] {"error":{"root_cause":[{"type":"query_shard_exception","reason":"Failed to parse query [AND gems]","index":"rubygems-development_20230823213447323","index_uuid":"9nPejNFqRICxsZW88Ub-WA"}],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[{"shard":0,"index":"rubygems-development_20230823213447323","node":"Q-0bFUllT-qQi5dQaMV_Jw","reason":{"type":"query_shard_exception","reason":"Failed to parse query [AND gems]","index":"rubygems-development_20230823213447323","index_uuid":"9nPejNFqRICxsZW88Ub-WA","caused_by":{"type":"parse_exception","reason":"Cannot parse 'AND gems': Encountered \" <AND> \"AND \"\" at line 1, column 0.\nWas expecting one of:\n    <NOT> ...\n    \"+\" ...\n    \"-\" ...\n    <BAREOPER> ...\n    \"(\" ...\n    \"*\" ...\n    <QUOTED> ...\n    <TERM> ...\n    <PREFIXTERM> ...\n    <WILDTERM> ...\n    <REGEXPTERM> ...\n    \"[\" ...\n    \"{\" ...\n    <NUMBER> ...\n    <TERM> ...\n    ","caused_by":{"type":"parse_exception","reason":"Encountered \" <AND> \"AND \"\" at line 1, column 0.\nWas expecting one of:\n    <NOT> ...\n    \"+\" ...\n    \"-\" ...\n    <BAREOPER> ...\n    \"(\" ...\n    \"*\" ...\n    <QUOTED> ...\n    <TERM> ...\n    <PREFIXTERM> ...\n    <WILDTERM> ...\n    <REGEXPTERM> ...\n    \"[\" ...\n    \"{\" ...\n    <NUMBER> ...\n    <TERM> ...\n    "}}}}]},"status":400}>
```

Reply:
the underlying `parse_exception` has a more descriptive error message in it, looks like

Reply:
That's seems to much "technical" IMHO, but anyway there is probably no simple way to get this JSON without parsing whole message (including leading `[400]`) string. There is no other method implemented keeping original response (https://github.com/opensearch-project/opensearch-ruby/blob/cea927498ff6c864c2427ed4c9e20bded15db0e4/lib/opensearch/transport/transport/errors.rb).

```
irb(#<ElasticSearcher:0x00007fe22946b348>):025:0> e
=> #<Searchkick::InvalidQueryError: [400] {"error":{"root_cause":[{"type":"query_shard_exception","reason":"Failed to parse query [AND gems]","index":"rubygems-development_20230823213447323","index_uuid":"9nPejNFqRICxsZW88Ub-WA"}],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[{"shard":0,"index":"rubygems-development_20230823213447323","node":"Q-0bFUllT-qQi5dQaMV_Jw","reason":{"type":"query_shard_exception","reason":"Failed to parse query [AND gems]","index":"rubygems-development_20230823213447323","index_uuid":"9nPejNFqRICxsZW88Ub-WA","caused_by":{"type":"parse_exception","reason":"Cannot parse 'AND gems': Encountered \" <AND> \"AND \"\" at line 1, column 0.\nWas expecting one of:\n    <NOT> ...\n    \"+\" ...\n    \"-\" ...\n    <BAREOPER> ...\n    \"(\" ...\n    \"*\" ...\n    <QUOTED> ...\n    <TERM> ...\n    <PREFIXTERM> ...\n    <WILDTERM> ...\n    <REGEXPTERM> ...\n    \"[\" ...\n    \"{\" ...\n    <NUMBER> ...\n    <TERM> ...\n    ","caused_by":{"type":"parse_exception","reason":"Encountered \" <AND> \"AND \"\" at line 1, column 0.\nWas expecting one of:\n    <NOT> ...\n    \"+\" ...\n    \"-\" ...\n    <BAREOPER> ...\n    \"(\" ...\n    \"*\" ...\n    <QUOTED> ...\n    <TERM> ...\n    <PREFIXTERM> ...\n    <WILDTERM> ...\n    <REGEXPTERM> ...\n    \"[\" ...\n    \"{\" ...\n    <NUMBER> ...\n    <TERM> ...\n    "}}}}]},"status":400}>
irb(#<ElasticSearcher:0x00007fe22946b348>):026:0> e.message
=> "[400] {\"error\":{\"root_cause\":[{\"type\":\"query_shard_exception\",\"reason\":\"Failed to parse query [AND gems]\",\"index\":\"rubygems-development_20230823213447323\",\"index_uuid\":\"9nPejNFqRICxsZW88Ub-WA\"}],\"type\":\"search_phase_execution_exception\",\"reason\":\"all shards failed\",\"phase\":\"query\",\"grouped\":true,\"failed_shards\":[{\"shard\":0,\"index\":\"rubygems-development_20230823213447323\",\"node\":\"Q-0bFUllT-qQi5dQaMV_Jw\",\"reason\":{\"type\":\"query_shard_exception\",\"reason\":\"Failed to parse query [AND gems]\",\"index\":\"rubygems-development_20230823213447323\",\"index_uuid\":\"9nPejNFqRICxsZW88Ub-WA\",\"caused_by\":{\"type\":\"parse_exception\",\"reason\":\"Cannot parse 'AND gems': Encountered \\\" <AND> \\\"AND \\\"\\\" at line 1, column 0.\\nWas expecting one of:\\n    <NOT> ...\\n    \\\"+\\\" ...\\n    \\\"-\\\" ...\\n    <BAREOPER> ...\\n    \\\"(\\\" ...\\n    \\\"*\\\" ...\\n    <QUOTED> ...\\n    <TERM> ...\\n    <PREFIXTERM> ...\\n    <WILDTERM> ...\\n    <REGEXPTERM> ...\\n    \\\"[\\\" ...\\n    \\\"{\\\" ...\\n    <NUMBER> ...\\n    <TERM> ...\\n    \",\"caused_by\":{\"type\":\"parse_exception\",\"reason\":\"Encountered \\\" <AND> \\\"AND \\\"\\\" at line 1, column 0.\\nWas expecting one of:\\n    <NOT> ...\\n    \\\"+\\\" ...\\n    \\\"-\\\" ...\\n    <BAREOPER> ...\\n    \\\"(\\\" ...\\n    \\\"*\\\" ...\\n    <QUOTED> ...\\n    <TERM> ...\\n    <PREFIXTERM> ...\\n    <WILDTERM> ...\\n    <REGEXPTERM> ...\\n    \\\"[\\\" ...\\n    \\\"{\\\" ...\\n    <NUMBER> ...\\n    <TERM> ...\\n    \"}}}}]},\"status\":400}"
irb(#<ElasticSearcher:0x00007fe22946b348>):027:0> e.cause
=> #<OpenSearch::Transport::Transport::Errors::BadRequest: [400] {"error":{"root_cause":[{"type":"query_shard_exception","reason":"Failed to parse query [AND gems]","index":"rubygems-development_20230823213447323","index_uuid":"9nPejNFqRICxsZW88Ub-WA"}],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[{"shard":0,"index":"rubygems-development_20230823213447323","node":"Q-0bFUllT-qQi5dQaMV_Jw","reason":{"type":"query_shard_exception","reason":"Failed to parse query [AND gems]","index":"rubygems-development_20230823213447323","index_uuid":"9nPejNFqRICxsZW88Ub-WA","caused_by":{"type":"parse_exception","reason":"Cannot parse 'AND gems': Encountered \" <AND> \"AND \"\" at line 1, column 0.\nWas expecting one of:\n    <NOT> ...\n    \"+\" ...\n    \"-\" ...\n    <BAREOPER> ...\n    \"(\" ...\n    \"*\" ...\n    <QUOTED> ...\n    <TERM> ...\n    <PREFIXTERM> ...\n    <WILDTERM> ...\n    <REGEXPTERM> ...\n    \"[\" ...\n    \"{\" ...\n    <NUMBER> ...\n    <TERM> ...\n    ","caused_by":{"type":"parse_exception","reason":"Encountered \" <AND> \"AND \"\" at line 1, column 0.\nWas expecting one of:\n    <NOT> ...\n    \"+\" ...\n    \"-\" ...\n    <BAREOPER> ...\n    \"(\" ...\n    \"*\" ...\n    <QUOTED> ...\n    <TERM> ...\n    <PREFIXTERM> ...\n    <WILDTERM> ...\n    <REGEXPTERM> ...\n    \"[\" ...\n    \"{\" ...\n    <NUMBER> ...\n    <TERM> ...\n    "}}}}]},"status":400}>
irb(#<ElasticSearcher:0x00007fe22946b348>):028:0> e.cause.message
=> "[400] {\"error\":{\"root_cause\":[{\"type\":\"query_shard_exception\",\"reason\":\"Failed to parse query [AND gems]\",\"index\":\"rubygems-development_20230823213447323\",\"index_uuid\":\"9nPejNFqRICxsZW88Ub-WA\"}],\"type\":\"search_phase_execution_exception\",\"reason\":\"all shards failed\",\"phase\":\"query\",\"grouped\":true,\"failed_shards\":[{\"shard\":0,\"index\":\"rubygems-development_20230823213447323\",\"node\":\"Q-0bFUllT-qQi5dQaMV_Jw\",\"reason\":{\"type\":\"query_shard_exception\",\"reason\":\"Failed to parse query [AND gems]\",\"index\":\"rubygems-development_20230823213447323\",\"index_uuid\":\"9nPejNFqRICxsZW88Ub-WA\",\"caused_by\":{\"type\":\"parse_exception\",\"reason\":\"Cannot parse 'AND gems': Encountered \\\" <AND> \\\"AND \\\"\\\" at line 1, column 0.\\nWas expecting one of:\\n    <NOT> ...\\n    \\\"+\\\" ...\\n    \\\"-\\\" ...\\n    <BAREOPER> ...\\n    \\\"(\\\" ...\\n    \\\"*\\\" ...\\n    <QUOTED> ...\\n    <TERM> ...\\n    <PREFIXTERM> ...\\n    <WILDTERM> ...\\n    <REGEXPTERM> ...\\n    \\\"[\\\" ...\\n    \\\"{\\\" ...\\n    <NUMBER> ...\\n    <TERM> ...\\n    \",\"caused_by\":{\"type\":\"parse_exception\",\"reason\":\"Encountered \\\" <AND> \\\"AND \\\"\\\" at line 1, column 0.\\nWas expecting one of:\\n    <NOT> ...\\n    \\\"+\\\" ...\\n    \\\"-\\\" ...\\n    <BAREOPER> ...\\n    \\\"(\\\" ...\\n    \\\"*\\\" ...\\n    <QUOTED> ...\\n    <TERM> ...\\n    <PREFIXTERM> ...\\n    <WILDTERM> ...\\n    <REGEXPTERM> ...\\n    \\\"[\\\" ...\\n    \\\"{\\\" ...\\n    <NUMBER> ...\\n    <TERM> ...\\n    \"}}}}]},\"status\":400}"

```